src/main/java/com/ververica/flink/table/gateway/operation/AbstractJobOperation.java

```diff
@@ -79,24 +70,14 @@ public AbstractJobOperation(SessionContext context) {
 		this.bufferedChangeFlags = null;
 		this.noMoreResults = false;
 		this.isJobCanceled = false;
+		this.clusterDescriptorAdapter =
+				ClusterDescriptorAdapterFactory.create(context.getExecutionContext(), sessionId);
```

Comment:
Build `ClusterDescriptorAdapter` after the job is submitted. It will only be used after that.